unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -40,6 +42,12 @@ private void Awake()
             currentPlayerInfoCardId = Resources.Load<StringVariable>(CURRENT_PLAYER_ID);
         }
 
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            StartCoroutine(SetFacialFeaturesVisibleRoutine());
+        }
```

Comment:
Should we use the `CourutineStarter ` here? Also, we should track the coroutine and clean it in the Cleanup method

Reply:
`CoroutineStarter` is a "hack" to have coroutines when you dont have Monobehaviours, which is not the case.

Tracking the coroutine is not really needed because they are aborted when the monobehaviour gets disabled (int his case, when going back to the pool), but it doesn't hurt to make it explicit. Gonna do it!

Reply:
Cool! As far as I know, we were using `CoroutineStarter ` (Besides de hack of course) because in some situations the coroutines will stay in memory even if the monobehaviour is destroyed. @BrianAmadori  can you confirm on this?

Reply:
I created the `CoroutineStarter` and that was the purpose, to have coroutines non dependant of monobehaviours haha.

Coroutines are parts of the monobehaviours (that's the reason if you have a disabled one you cannot start a coroutine, because the update is never called) and I haven't been able to find anything related to leaks in coroutines. 
In fact using the `CoroutineStarter` can actually lead to some leaks if the coroutine it's never released and the handler gets destroyed.